monitor/src/main/java/edu/asu/diging/monitor/core/service/impl/AppHelper.java

```diff
@@ -95,5 +103,12 @@ private List<NotificationRecipient> getRecipientsById(List<String> recipientIds)
             return (NotificationRecipient) manager.getRecipient(id);
         }).collect(Collectors.toList());
     }
+    
+    private List<Tag> getTagsByName(List<String> tagNames){
+        return tagNames.stream().map(t -> {
+            Tag temp = (Tag) tagManager.getTagByName(t);
+            return (temp != null) ? temp : new Tag(t);
```

Comment:
if there is no tag with the given label, should it really create a new one on the fly? when is this used? and if this is the desired functionality, the method name should be different as it implies this method gets tags from the database, not create them.

Reply:
This method is  indirectly called from the addApp/updateApp method in AppManager. The AppForm has the list of tags as Strings (tag names) and the new and existing tags are all submitted by the view in the same format. I'm using this method to fetch the tags if they already exist in the DB/ create new tags if there is no instance of the tag currently in the DB. I'll change the name of the method to better reflect what its doing